mm/umm_heap/umm_memalign.c

```diff
@@ -54,6 +54,10 @@ FAR void *memalign(size_t alignment, size_t size)
   FAR void *brkaddr;
   FAR void *mem;
 
+  /* Initialize the user heap if not yet */
+
+  umm_try_initialize();
+
```

Comment:
```suggestion
  /* Initialize the user heap if wasn't yet. */

  umm_try_initialize();

```

Why is this required here?

Reply:
Kernel mode doesn't call umm_initialize in nx_start(it's impossible from technique since each process has a dedicated heap instance). Before this patch, sbrk will initialize the heap when malloc fail first time and fallback to sbrk, but we have to initialize explicitly here because tg_heap change to pointer and then no place to save the flag before we let g_heap point to some memory.

Reply:
the comment is fixed, @Ouss4

Reply:
Thanks for the explanation!